xarray/test/test_dataset.py

```diff
@@ -2263,6 +2263,25 @@ def test_fillna(self):
         actual = ds.groupby('b').fillna(Dataset({'a': ('b', [0, 2])}))
         self.assertDatasetIdentical(expected, actual)
 
+        # attrs with groupby
+        ds.attrs['attr'] = 'ds'
+        ds.a.attrs['attr'] = 'da'
+        actual = ds.groupby('b').fillna(Dataset({'a': ('b', [0, 2])}))
+        self.assertEqual(actual.attrs, ds.attrs)
+        self.assertTrue(actual.a.name == 'a')
```

Comment:
this is what happens when one does too many things at the same time without thinking :-(

Should be ok now, hopefully...